server/src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -82,11 +85,22 @@ public void storeMessage(Message message) {
 
     return messages;
   }
-  
+
   /** Returns the total number of messages for all users. */
   public int getTotalMessageCount() {
     Query query = new Query("Message");
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withLimit(MESSAGE_LIMIT));
   }
+
+  /** Returns a list of all the users that have sent messages. */
+  public Set<String> getUsers() {
```

Comment:
To be honest, it might be more interesting to see all the users that have signed up. It's up to you, but if you choose to show all the users who have signed up then you'll need to add some logic in the LoginServlet.

Reply:
That's a good idea. I'm going to work on that and leave it for another pull request